src/bun.js/node/node_cluster_binding.zig

```diff
@@ -0,0 +1,275 @@
+const std = @import("std");
+const bun = @import("root").bun;
+const Environment = bun.Environment;
+const JSC = bun.JSC;
+const string = bun.string;
+const Output = bun.Output;
+const ZigString = JSC.ZigString;
+const log = Output.scoped(.IPC, false);
+
+extern fn Bun__Process__queueNextTick1(*JSC.ZigGlobalObject, JSC.JSValue, JSC.JSValue) void;
+extern fn Process__emitErrorEvent(global: *JSC.JSGlobalObject, value: JSC.JSValue) void;
+
+pub var child_singleton: InternalMsgHolder = .{};
+
+pub fn sendHelperChild(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(JSC.conv) JSC.JSValue {
+    log("sendHelperChild", .{});
+
+    const arguments = callframe.arguments(3).ptr;
+    const message = arguments[0];
+    const handle = arguments[1];
+    const callback = arguments[2];
+
+    const vm = globalThis.bunVM();
+
+    if (vm.ipc == null) {
+        return .false;
+    }
+    if (message.isUndefined()) {
+        return globalThis.throwValueRet(globalThis.ERR_MISSING_ARGS_static(ZigString.static("message"), null, null));
+    }
+    if (!handle.isNull()) {
+        globalThis.throw("passing 'handle' not implemented yet", .{});
+        return .zero;
+    }
+    if (!message.isObject()) {
+        return globalThis.throwValueRet(globalThis.ERR_INVALID_ARG_TYPE_static(
+            ZigString.static("message"),
+            ZigString.static("object"),
+            message,
+        ));
+    }
+    if (callback.isFunction()) {
+        child_singleton.callbacks.put(bun.default_allocator, child_singleton.seq, JSC.Strong.create(callback, globalThis)) catch bun.outOfMemory();
+    }
+
+    message.put(globalThis, ZigString.static("seq"), JSC.JSValue.jsNumber(child_singleton.seq));
+    child_singleton.seq +%= 1;
+
+    // similar code as Bun__Process__send
+    var formatter = JSC.ConsoleObject.Formatter{ .globalThis = globalThis };
+    if (Environment.isDebug) log("child: {}", .{message.toFmt(&formatter)});
+
+    const ipc_instance = vm.getIPCInstance().?;
+    const process_queueNextTick1 = Bun__Process__queueNextTick1;
+    const zigGlobal: *JSC.ZigGlobalObject = @ptrCast(globalThis);
+
+    const S = struct {
+        fn impl(globalThis_: *JSC.JSGlobalObject, callframe_: *JSC.CallFrame) callconv(JSC.conv) JSC.JSValue {
+            const arguments_ = callframe_.arguments(1).slice();
+            const ex = arguments_[0];
+            Process__emitErrorEvent(globalThis_, ex);
+            return .undefined;
+        }
+        var value: ?JSC.JSValue = null;
```

Comment:
We cannot hold a JSValue this way. The GC won't see it. It's a global variable for something which is inherently thread-local.